client/consensus/src/lib.rs

```diff
@@ -145,14 +148,13 @@ use {
 /// and makes no guarantees which one as that depends on the keystore's iterator behavior.
 /// This is the standard way of determining which key to author with.
 /// It also returns its ParaId assignment
-pub fn first_eligible_key<B: BlockT, C, P>(
+pub async fn first_eligible_key<C, P>(
     client: &C,
-    parent_hash: &B::Hash,
+    parent_hash: &H256,
```

Comment:
why this change? isnt this forcing us to use one specific hash?

Reply:
Removing the first generic I mean

Reply:
This fixes a compile error, I guess now we are forced to always use H256.

Caused by `check_para_id_assignment` function requiring H256.

Reply:
can we change `check_para_id_assignment` to accept a generic? or is it too much of a change? cause this feels "bad" ina  way

Reply:
We can if we make the `OrchestratorChainInterface` trait generic, but I think it's a bad idea.

Reply:
Alright, let's leave it as is for ow..

Reply:
disregard